### PR TITLE
Replace node-spellchecker with Electron builtin spell checker

### PR DESCRIPTION
**Pull request by fxha**
| Q                 | A
| ----------------- | ---
| Breaking changes? | yes
| License           | MIT

### Description

This PR tries to replace `node-spellchecker` with Electrons built-in spell checker to get the latest spelling engine and to reduce maintenance of the spell checker module.

**To-do's:**

- [x] Unable to enable spell checker if it was disabled at window launch (https://github.com/electron/electron/issues/32755)

**Breaking changes:**

- No spelling options are available on macOS, like switching language, because all settings are managed by the OS. Language is automatically detected while typing as before on macOS.
- Spelling language cannot longer set per editor window
- Ignoring a word at runtime is no longer possible.

### CODE REVIEW

**Comment by fxha** [comment]
@Jocs Please test this draft on macOS, whether

1. misspelled words are marked in editor
2. replacing a misspelled word works correctly
3. adding a misspelled word to dictionary works
  - is still present after restart of MarkText
  - "Edit Dictionary" in context menu prints the added words
4. enabling and disabling the spell checker is fine 
5. etc

**Comment by Jocs** [comment]
@fxha I'll test it this weekend(today or tomorrow).

**Comment by Jocs** [comment]
@fxha Please fix the conflicts.

**Comment by Jocs** [comment]
@fxha Two problems when I test on macOS

1. Click "Edit Dictionary" in context menu, and nothing happened, what would happen when I click this menu?
2. Disable `Spell checker` works well, but the wrong word is not marked red dots underline when I enable `Spell checker`.

**Comment by fxha** [comment]
I planned the PR for v0.18 because a few things are missing and there were multiple upstream issues on Windows. 

> Click "Edit Dictionary" in context menu, and nothing happened, what would happen when I click this menu?

This isn't yet implemented.

> Disable Spell checker works well, but the wrong word is not marked red dots underline when I enable Spell checker.

I notice the same on Windows, but it worked if you type text and wait a bit. Seems to be an Chromium issue that the spell checker is only invoked if the website content changes.

**Comment by Jocs** [comment]
>I planned the PR for v0.18 because a few things are missing and there were multiple upstream issues on Windows.

OK.

**Comment by Jocs** [comment]
@fxha I merged the `update Electron version` PR, and conflict happened.

**Comment by fxha** [comment]
> It seems not work on macOS system, the misspelled words are not marked in editor.

Have you tried to disable and re-enable spelling in settings? It might be possible that we need to manually enable the spellchecker via API as we do on Linux/Windows. Most likely an issue because macOS spellchecker doesn't require a language to work.

Edit: I might found the issue and will summit a patch soon.

File: src/muya/lib/utils/index.js [hunk 1]
```diff
@@ -433,3 +434,12 @@ export const getDefer = () => {
 
   return defer
 }
+
+/**
+ * Deep clone the given object.
+ *
+ * @param {*} obj Object to clone
+ */
+export const deepClone = obj => {
```
[the remark this answers is not in the document]
Comment (fxha) [reply src/muya/lib/utils/index.js:443]: I cannot find another one in this file?

**Comment by Jocs** [comment]
>Have you tried to disable and re-enable spelling in settings? It might be possible that we need to manually enable the spellchecker via API as we do on Linux/Windows. Most likely an issue because macOS spellchecker doesn't require a language to work.

I found I need to change macOS system language to English to enable English spellchecker.

**Comment by Jocs** [comment]
>Have you tried to disable and re-enable spelling in settings? It might be possible that we need to manually enable the spellchecker via API as we do on Linux/Windows. Most likely an issue because macOS spellchecker doesn't require a language to work.

I found I need to change macOS system language to English to enable English spellchecker.

**Review by Jocs — APPROVED** [review]

**Comment by fxha** [comment]
> I found I need to change macOS system language to English to enable English spellchecker

Do we need to add/update documentation in settings?

**Pull request merged**